### DISCUSSION TITLE
IE 11 Support?

### DISCUSSION THREAD

**Discussion by jwohlfert23** [livewire/livewire#3093] [7. Bugs (archived)]
Using inline scripts in IE 11 doesn't seem to work.  

For example, this script:
```html
<script>
 document.addEventListener('livewire:load', function() {
        document.getElementById('payment-form').addEventListener('submit', function (ev) {
              ev.preventDefault();
              @this.submit(); 
        });
});
</script>
```
Produces the following error:
`Object doesn't support property or method 'submit'`

This same error occurs when trying to use `@entangle`.  It seems like any time a method is being called via an inline script on a livewire component, it produces this error.

`window.livewire.find('componentId').anyMethodHere()` => `Object doesn't support property or method 'anyMethodHere'`

I've tried pretty much anything you can think of:
- Including ES5/ES6 Polyfills
- With/Without Alpine
- Loading Alpine Before/After

**Context**
- Livewire version: 2.3.5
- Laravel version: 8.20.1
- Alpine version: 2.8.0
- Browser: Internet Explorer 11

**Comment by jwohlfert23** [comment]
After doing some more digging, it appears this is likely due to the fact that the `$wire` property of a Livewire component is implemented using a `Proxy` which isn't supported in IE 11.

Looks like it's a difficult thing to polyfill as well. Does anyone have any leads on polyfills or workarounds to this issue?  Not being able to use Inline Scripts if you need to support IE 11 seems like a pretty big issue.

**Comment by marispro** [comment]
+1
Also some project required IE 11, and for my surprise this is only thing that doesn't work.
Not sure should it be pointed to Livewire or Alpine in case of entangle. :)

![Screenshot_1](https://user-images.githubusercontent.com/5610414/104330829-b1fc8c80-54ee-11eb-9e11-afdfd12cc51a.png)
.

**Comment by joshhanley** [comment]
@marispro yeah it's a Livewire issue, as @jwohlfert23 pointed out, $wire is a Livewire object that is a proxy and window.Livewire.find just returns the underlying $wire proxy.

@jwohlfert23 when you say you tried polyfills, how did you try it? Did you compile Livewire's javascript assets from the source or try to run a polyfill over the compiled assets?

**Comment by jwohlfert23** [comment]
I ended up finding a workaround for this without having to recompile Livewire's JS:

```js
import ProxyPolyfillBuilder from 'proxy-polyfill';

const proxyProps = [
  'entangle',
  ...otherFunctionNamesYouWantToCallInJS
];

const ProxyPolyfill = ProxyPolyfillBuilder();
window.Proxy = function (target, handler) {
  if (!Array.isArray(target)) {
    proxyProps.forEach(function (key) {
      if (!target.hasOwnProperty(key)) {
        Object.assign(target, { [key]: null });
      }
    });
  }
  return new ProxyPolyfill(target, handler);
};
```
In summary, `proxy-polyfill` only works if you know which properties/functions are going to be called on a proxy **at creation time**.  So in the above script, I'm just setting each of those properties to null on the proxy target, so the polyfill knows which properties I need.  This script is then included using the same approach AlpineJS does, so it only gets loaded on IE.

```html
<script nomodule src="{{mix('/js/ie11.js')}}"></script>
```

It's not a great solution, but it works.  In my opinion, this should be included in Livewire or IE 11 support should be dropped, because inline scripts 100% do not work in IE 11 right now as is.
